prac_07/project_management.py

```diff
@@ -0,0 +1,122 @@
+import csv
+from datetime import datetime
+from project import Project
+
+FILENAME = "projects.txt"
+
+
+def main():
+    projects = load_projects(FILENAME)
+    print(f"Welcome to Pythonic Project Management\nLoaded {len(projects)} projects from {FILENAME}")
+    menu = """
+    - (L)oad projects  
+    - (S)ave projects  
+    - (D)isplay projects  
+    - (F)ilter projects by date
+    - (A)dd new project  
+    - (U)pdate project
+    - (Q)uit
+    """
+
+    choice = ""
+    while choice.lower() != 'q':
+        print(menu)
+        choice = input(">>> ").lower()
+        if choice == 'l':
+            filename = input("Enter filename to load: ")
+            projects = load_projects(filename)
+        elif choice == 's':
+            filename = input("Enter filename to save: ")
+            save_projects(filename, projects)
+        elif choice == 'd':
+            display_projects(projects)
+        elif choice == 'f':
+            filter_projects_by_date(projects)
+        elif choice == 'a':
+            add_project(projects)
+        elif choice == 'u':
+            update_project(projects)
+
+    if input(f"Would you like to save to {FILENAME}? (y/n): ").lower() == 'y':
+        save_projects(FILENAME, projects)
+    print("Thank you for using custom-built project management software.")
+
+
+def load_projects(filename):
+    """Load projects from projects.txt return list of Project objects"""
+    projects = []
+    with open(filename, 'r') as file:
+        reader = csv.reader(file, delimiter='\t')
+        next(reader)  # Skip header
+        for row in reader:
+            projects.append(Project(*row))
+    return projects
+
+
+def save_projects(filename, projects):
+    """Save all projects to file"""
+    with open(filename, 'w', newline='') as file:
+        writer = csv.writer(file, delimiter='\t')
+        writer.writerow(["Name", "Start Date", "Priority", "Cost Estimate", "Completion Percentage"])
+        for project in projects:
+            writer.writerow(
+                [project.name, project.start_date.strftime("%d/%m/%Y"), project.priority, project.cost_estimate,
+                 project.completion_percentage])
+
+
+def display_projects(projects):
+    """display incomplete and completed projects, sorted by priority"""
```

Comment:
Noticed a few inconsistencies with the docstrings not always being in the correct format (capital start and full stop at the end)